exporter/stackdriverexporter/stackdriver.go

```diff
@@ -76,6 +76,11 @@ func generateClientOptions(cfg *Config, dialOpts ...grpc.DialOption) ([]option.C
 	} else {
 		copts = append(copts, option.WithEndpoint(cfg.Endpoint))
 	}
+	if len(cfg.ClientOptions) > 0 {
```

Comment:
`generateClientOptions` is not called from all code paths. It is only called if `cfg.Endpoint != ""` (which is currently the case for all the tests as we don't have a mock framework at the moment)

I think it's probably worth refactoring some of the code surrounding the two calls to this function and bringing it into this function.